geoh5py/ui_json/parameters.py

```diff
@@ -0,0 +1,491 @@
+
+    @property
+    def active(self):
+        active = [k[1:] for k, v in self.__dict__.items() if isinstance(v, Parameter)]
+        return active + list(self._extra_members)
+
+    @property
+    def required(self):
+        return [k for k, v in self.form_validations.items() if v.get("required", False)]
+
+    @property
+    def form(self):
+        form = {}
+        for member in self.active:
+            if member in self._extra_members:
+                form[member] = self._extra_members[member]
+            else:
+                form[member] = getattr(self, member)
+
+        return form
+
+    def validate(self):
+        for member in set(self.active + self.required):
+            try:
+                getattr(self, f"_{member}").validate()
+            except BaseValidationError as err:
+                raise UIJsonFormatError(self.name, str(err)) from err
+
+    @classmethod
+    def is_form(cls, form: dict[str, Any]) -> bool:
+        id_members = cls.identifier_members
+        form_members = [cls.key_map.get(k, k) for k in form]
+        return any(k in form_members for k in id_members)
+
+    def __str__(self):
+        return f"<{type(self).__name__}> : '{self.name}' -> {self.value}"
+
+    def __getattr__(self, name):
+        if f"_{name}" in self.__dict__:
+            member = self.__dict__[f"_{name}"]
+        elif name == "validations":
+            member = self.__dict__["_value"].validations
+        else:
+            try:
+                member = self.__dict__[name]
+            except KeyError as err:
+                raise AttributeError(f"'{name}' attribute doesn't exist.") from err
+
+        return member.value if isinstance(member, Parameter) else member
+
+    def __setattr__(self, name, value):
+        if name in self.valid_members:
+            self.__dict__[f"_{name}"] = Parameter(
+                name, value, self.form_validations[name]
+            )
+        elif name == "validations":
+            self.__dict__["_value"].validations = value
+        else:
+            self.__dict__[name] = value
+
+
+class StringParameter(FormParameter):
+    base_validations: Validation = {"types": [str]}
+
+    def __init__(self, name, validations=None, **kwargs):
+        validations = (
+            dict(self.base_validations, **validations)
+            if validations
+            else self.base_validations
+        )
+        super().__init__(name, validations=validations, **kwargs)
+
+
+class BoolParameter(FormParameter):
+    base_validations: Validation = {"types": [bool]}
+
+    def __init__(self, name, validations=None, **kwargs):
+        validations = (
+            dict(self.base_validations, **validations)
+            if validations
+            else self.base_validations
+        )
+        super().__init__(name, validations=validations, **kwargs)
+
+
+class IntegerParameter(FormParameter):
+    base_validations: Validation = {"types": [int]}
+    integer_form_validations: dict[str, Validation] = {

```

Comment:
Are you going to refer to this class attribute later? If not, I would just set the form_validations directly.

Reply:
done